libsolidity/ast/Types.cpp

```diff
@@ -3499,7 +3499,9 @@ bool FunctionType::isPure() const
 		m_kind == Kind::ABIEncodeWithSelector ||
 		m_kind == Kind::ABIEncodeWithSignature ||
 		m_kind == Kind::ABIDecode ||
-		m_kind == Kind::MetaType;
+		m_kind == Kind::MetaType ||
+		m_kind == Kind::Wrap ||
+		m_kind == Kind::Unwrap;
```

Comment:
The `FunctionType` already has the state mutability set to `Pure` (see https://github.com/ethereum/solidity/blob/develop/libsolidity/ast/Types.cpp#L3813), but turns out that seem to be ignored. There seems to be a TODO on top to fix this.

Reply:
These might be the two different concepts of pureness in the compiler.